### PR TITLE
refactor: address instrumentation-http TODOs

### PR DESCRIPTION
**Pull request by fabiogomessilva**
This PR address instrumentation-http TODOs

### CODE REVIEW

File: packages/opencensus-instrumentation-http/src/http.ts [hunk 1]
```diff
@@ -290,15 +285,15 @@ export class HttpPlugin extends classes.BasePlugin {
         response.on('end', () => {
           plugin.logger.debug('outgoingRequest on end()');
           const method = response.method ? response.method : 'GET';
-          const reqUrl = url.parse(options);
           const userAgent =
-              headers ? (headers['user-agent'] || headers['User-Agent']) : null;
+              headers ? (headers['user-agent'] || headers['User-Agent']) : '';
```
Comment (kjin) [inline packages/opencensus-instrumentation-http/src/http.ts]: If there is no user agent... will it append an empty label? We should make the label conditional on there being a user agent.

File: packages/opencensus-instrumentation-http/src/http.ts [hunk 2]
```diff
@@ -24,7 +24,6 @@ import * as url from 'url';
 import * as uuid from 'uuid';
 
 
-// TODO: maybe we should have a setup as a Client or as Server.
```
Comment (kjin) [inline packages/opencensus-instrumentation-http/src/http.ts]: Has this TODO been addressed? I don't see any change that reflects that it has.

File: packages/opencensus-instrumentation-http/src/http.ts [hunk 3]
```diff
@@ -80,7 +78,7 @@ export class HttpPlugin extends classes.BasePlugin {
       shimmer.wrap(
           moduleExports && moduleExports.Server &&
               moduleExports.Server.prototype,
-          'emit', this.patchIncomingRequest());
+          'emit' as never, this.patchIncomingRequest());
```
Comment (kjin) [inline packages/opencensus-instrumentation-http/src/http.ts]: Why this change?

**Comment by kjin** [comment]
While you're on this file, can you also make `patch*` functions private?

File: packages/opencensus-instrumentation-http/src/http.ts (hunk 2 above)
Comment (fabiogomessilva) [reply packages/opencensus-instrumentation-http/src/http.ts]: Yes, there is no need to have specific setups for client and servers.  No action to do.

File: packages/opencensus-instrumentation-http/src/http.ts (hunk 3 above)
Comment (fabiogomessilva) [reply packages/opencensus-instrumentation-http/src/http.ts]: This change is a workaround to solve a strange type cast error for this.patchIncomingRequest when we set the type for moduleExports as HttpModule.

Comment (fabiogomessilva) [reply packages/opencensus-instrumentation-http/src/http.ts]: Regarding method visibility, we can not set patch* as private because subclass https needs access to those methods - they were set to protected instead.

File: packages/opencensus-instrumentation-http/src/http.ts (hunk 1 above)
Comment (fabiogomessilva) [reply packages/opencensus-instrumentation-http/src/http.ts]: Done!

File: packages/opencensus-instrumentation-http/src/http.ts (hunk 3 above)
Comment (kjin) [reply packages/opencensus-instrumentation-http/src/http.ts]: Would this work?

```js
shimmer.wrap(moduleExports.Server.prototype, 'emit', /*...*/);
```

Comment (fabiogomessilva) [reply packages/opencensus-instrumentation-http/src/http.ts]: Could not make it work.

Comment (kjin) [reply packages/opencensus-instrumentation-http/src/http.ts]: Ok. Might be a problem with the shimmer type definitions.
